### PR TITLE
Prevent password reset code leakage via cross origin requests

### PR DESCRIPTION
**Pull request by ElizabethStirling**
Referrer header now only contains the origin URL, not the entire URL.

Resolves [Security Issue 74](https://github.com/sourcegraph/security-issues/issues/74)

### CODE REVIEW

File: cmd/frontend/internal/app/templates/ui/app.html [hunk 1]
```diff
@@ -7,7 +7,8 @@
 	<meta http-equiv="x-ua-compatible" content="ie=edge">
 	<meta name="google" content="notranslate">
 	<meta http-equiv="Content-Language" content="en">
-	<meta name="viewport" content="width=device-width, viewport-fit=cover" />
+    <meta name="viewport" content="width=device-width, viewport-fit=cover" />
```
Comment (slimsag) [inline cmd/frontend/internal/app/templates/ui/app.html:10]:
```suggestion
	<meta name="viewport" content="width=device-width, viewport-fit=cover" />
```

File: cmd/frontend/internal/app/templates/ui/app.html [hunk 2]
```diff
@@ -7,7 +7,8 @@
...
+    <meta name="referrer" content="origin-when-cross-origin"/>
```
Comment (slimsag) [inline cmd/frontend/internal/app/templates/ui/app.html:11]:
```suggestion
	<meta name="referrer" content="origin-when-cross-origin"/>
```

**Review by slimsag — APPROVED** [review]
LGTM - should also add a CHANGELOG entry perhaps?

**Review by chayim — APPROVED** [review]

**Review by tjkandala — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ All notable changes to Sourcegraph are documented in this file.
 - Fixed a bug where we returned repositories with invalid revisions in the search results. Now, if a user specifies an invalid revision, we show an alert. [#13271](https://github.com/sourcegraph/sourcegraph/pull/13271)
 - Previously it wasn't possible to search for certain patterns containing `:` because they would not be considered valid filters. We made these checks less strict. [#10920](https://github.com/sourcegraph/sourcegraph/pull/10920)
 - Logging out of a user's account will now invalidate all user sessions, not just the session that was signed out of. Resetting or changing the user's password will also invalidate all sesions, requiring the user to sign in with their new password. [#13647](https://github.com/sourcegraph/sourcegraph/pull/13647)
+- URL information will no longer be leaked by the HTTP referer header. This prevents the user's password reset code from being leaked. [#13804]https://github.com/sourcegraph/sourcegraph/pull/13804
```
Comment (unknwon) [inline CHANGELOG.md:45]: This seems to be a markdown syntax error :)

```diff
-[#13804]https://github.com/sourcegraph/sourcegraph/pull/13804
+[#13804](https://github.com/sourcegraph/sourcegraph/pull/13804)
```

**Pull request merged**